### ISSUE TITLE
Size of fp semigroup is negative

### ISSUE THREAD

**Issue by james-d-mitchell** [gap-packages/Semigroups#685]
~~~~
F := FreeSemigroup(2);
s1 := F.1; s2 := F.2;
rels := [ [ s2*s1*s2, s2*s1 ], [ s1, s1 ], [ s2, s2 ], [ s1*s2, s1*s2 ], [
        s2*s1, s2*s1 ] ];
Size(F / rels); # returns -2!!
~~~~

This is likely caused by `libsemigroups` returning that the fp semigroup is infinite, and this getting incorrectly returned as `-2` in GAP.

**Comment by alex-konovalov** [comment]
I wonder if `Size` in GAP should have an assertion to forbid negative sizes...

**Comment by james-d-mitchell** [comment]
Resolved in v3.4.0
